silkrpc/http/server.hpp

```diff
@@ -68,6 +68,7 @@ class Server {
     boost::asio::ip::tcp::acceptor acceptor_;
 
     boost::asio::thread_pool& workers_;
+    std::string jwt_secret_;
```

Comment:
In order to support both authenticated and unauthenticated `Server` instances, I suggest using `std::optional<std::string>` as type for `jwt_secret_` instead of using empty value: it expresses the programmer's intention much clearly